shared_model/validators/field_validator.hpp

```diff
@@ -36,39 +36,50 @@ namespace shared_model {
     class FieldValidator {
      public:
       FieldValidator()
-          : account_id_(R"([a-z]{1,9}\@[a-z]{1,9})"),
-            asset_id_(R"([a-z]{1,9}\#[a-z]{1,9})"),
-            ip_address_(
-                "((([0-1]?\\d\\d?)|((2[0-4]\\d)|(25[0-5]))).){3}(([0-1]?\\d\\d?"
-                ")|((2[0-4]\\d)|(25[0-5])))"),
-            name_(R"([a-z]{1,9})"),
-            detail_key_(R"([A-Za-z0-9_]{1,})") {}
+          : account_id_regex(R"([a-z]{1,9}\@[a-z]{1,9})"),
+            asset_id_regex(R"([a-z]{1,9}\#[a-z]{1,9})"),
+            name_regex(R"([a-z]{1,9})"),
+            detail_key_regex(R"([A-Za-z0-9_]{1,})") {}
 
       void validateAccountId(
           ReasonsGroupType &reason,
           const interface::types::AccountIdType &account_id) const {
-        if (not std::regex_match(account_id, account_id_)) {
-          reason.second.push_back("Wrongly formed account_id");
+        if (not std::regex_match(account_id, account_id_regex)) {
+          auto message =
+              (boost::format("Wrongly formed account_id, passed value: '%s'")
+               % static_cast<std::string>(account_id))
```

Comment:
It's not clear whether cast is required, since it seems unlikely that `AccountIdType` will have only explicit conversion to `std::string`. Same applies to other methods.